### ISSUE TITLE
Julia 1.4 v Julia 1.3 MPI resolving difference on OS X 

### ISSUE THREAD

**Issue by christophernhill** [JuliaParallel/MPI.jl#376]
Hi Gang,

 In case anyone else runs into this I am just switching to Julia 1.4 (```Julia Version 1.4.1```) on Mac ```macOS (x86_64-apple-darwin18.7.0)```. I seem to need to set 

```
ENV["JULIA_MPI_PATH"]="/usr/local/Cellar/open-mpi/4.0.3"
```

explicitly to avoid MPI not found. That does not seem to be the case in Julia 1.3.1

e.g.

```
\rm -fr ~/.julia  
/Applications/Julia-1.3.app/Contents/MacOS/applet

               _
   _       _ _(_)_     |  Documentation: https://docs.julialang.org
  (_)     | (_) (_)    |
   _ _   _| |_  __ _   |  Type "?" for help, "]?" for Pkg help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 1.3.1 (2019-12-30)
 _/ |\__'_|_|_|\__'_|  |  Official https://julialang.org/ release
|__/                   |

julia> using Pkg

julia> Pkg.add("MPI")
   Cloning default registries into `~/.julia`
   Cloning registry from "https://github.com/JuliaRegistries/General.git"
     Added registry `General` to `~/.julia/registries/General`
 Resolving package versions...
 Installed Requires ──────────── v1.0.1
 Installed DocStringExtensions ─ v0.8.1
 Installed MPI ───────────────── v0.13.1
  Updating `~/.julia/environments/v1.3/Project.toml`
  [da04e1cc] + MPI v0.13.1
  Updating `~/.julia/environments/v1.3/Manifest.toml`
  [ffbed154] + DocStringExtensions v0.8.1
  [da04e1cc] + MPI v0.13.1
  [ae029012] + Requires v1.0.1
  [2a0f44e3] + Base64 
  [ade2ca70] + Dates 
  [8ba89e20] + Distributed 
  [b77e0a4c] + InteractiveUtils 
  [76f85450] + LibGit2 
  [8f399da3] + Libdl 
  [56ddb016] + Logging 
  [d6f4376e] + Markdown 
  [44cfe95a] + Pkg 
  [de0858da] + Printf 
  [3fa0cd96] + REPL 
  [9a3f8284] + Random 
  [ea8e919c] + SHA 
  [9e88b42a] + Serialization 
  [6462fe0b] + Sockets 
  [8dfed614] + Test 
  [cf7118a7] + UUIDs 
  [4ec0a83e] + Unicode 
  Building MPI → `~/.julia/packages/MPI/ZfFyE/deps/build.log`

julia> 

```

```
 \rm -fr ~/.julia 
 /Applications/Julia-1.4.app/Contents/MacOS/applet

   _       _ _(_)_     |  Documentation: https://docs.julialang.org
  (_)     | (_) (_)    |
   _ _   _| |_  __ _   |  Type "?" for help, "]?" for Pkg help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 1.4.1 (2020-04-14)
 _/ |\__'_|_|_|\__'_|  |  Official https://julialang.org/ release
|__/                   |

julia> using Pkg

julia> Pkg.add("MPI")
    Cloning default registries into `~/.julia`
    Cloning registry from "https://github.com/JuliaRegistries/General.git"
      Added registry `General` to `~/.julia/registries/General`
  Resolving package versions...
  Installed Requires ──────────── v1.0.1
  Installed DocStringExtensions ─ v0.8.1
  Installed MPI ───────────────── v0.13.1
   Updating `~/.julia/environments/v1.4/Project.toml`
  [da04e1cc] + MPI v0.13.1
   Updating `~/.julia/environments/v1.4/Manifest.toml`
  [ffbed154] + DocStringExtensions v0.8.1
  [da04e1cc] + MPI v0.13.1
  [ae029012] + Requires v1.0.1
  [2a0f44e3] + Base64 
  [ade2ca70] + Dates 
  [8ba89e20] + Distributed 
  [b77e0a4c] + InteractiveUtils 
  [76f85450] + LibGit2 
  [8f399da3] + Libdl 
  [56ddb016] + Logging 
  [d6f4376e] + Markdown 
  [44cfe95a] + Pkg 
  [de0858da] + Printf 
  [3fa0cd96] + REPL 
  [9a3f8284] + Random 
  [ea8e919c] + SHA 
  [9e88b42a] + Serialization 
  [6462fe0b] + Sockets 
  [8dfed614] + Test 
  [cf7118a7] + UUIDs 
  [4ec0a83e] + Unicode 
   Building MPI → `~/.julia/packages/MPI/ZfFyE/deps/build.log`
┌ Error: Error building `MPI`: 
│ ERROR: LoadError: LoadError: No MPI library found.
│ Ensure an MPI implementation is loaded, or set the `JULIA_MPI_PATH` variable.
│ Stacktrace:
│  [1] error(::String) at ./error.jl:33
│  [2] find_lib() at /Users/chrishill/.julia/packages/MPI/ZfFyE/src/paths.jl:19
│  [3] top-level scope at /Users/chrishill/.julia/packages/MPI/ZfFyE/src/paths.jl:25
│  [4] include(::String) at ./client.jl:439
│  [5] top-level scope at /Users/chrishill/.julia/packages/MPI/ZfFyE/deps/build.jl:3
│  [6] include(::String) at ./client.jl:439
│  [7] top-level scope at none:5
│ in expression starting at /Users/chrishill/.julia/packages/MPI/ZfFyE/src/paths.jl:25
│ in expression starting at /Users/chrishill/.julia/packages/MPI/ZfFyE/deps/build.jl:3
└ @ Pkg.Operations /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.4/Pkg/src/Operations.jl:892

```

```
 \rm -fr ~/.julia 
 /Applications/Julia-1.4.app/Contents/MacOS/applet
              _
   _       _ _(_)_     |  Documentation: https://docs.julialang.org
  (_)     | (_) (_)    |
   _ _   _| |_  __ _   |  Type "?" for help, "]?" for Pkg help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 1.4.1 (2020-04-14)
 _/ |\__'_|_|_|\__'_|  |  Official https://julialang.org/ release
|__/                   |

julia> ENV["JULIA_MPI_PATH"]="/usr/local/Cellar/open-mpi/4.0.3"
"/usr/local/Cellar/open-mpi/4.0.3"

julia> using Pkg

julia> Pkg.add("MPI")
    Cloning default registries into `~/.julia`
    Cloning registry from "https://github.com/JuliaRegistries/General.git"
      Added registry `General` to `~/.julia/registries/General`
  Resolving package versions...
  Installed Requires ──────────── v1.0.1
  Installed DocStringExtensions ─ v0.8.1
  Installed MPI ───────────────── v0.13.1
   Updating `~/.julia/environments/v1.4/Project.toml`
  [da04e1cc] + MPI v0.13.1
   Updating `~/.julia/environments/v1.4/Manifest.toml`
  [ffbed154] + DocStringExtensions v0.8.1
  [da04e1cc] + MPI v0.13.1
  [ae029012] + Requires v1.0.1
  [2a0f44e3] + Base64 
  [ade2ca70] + Dates 
  [8ba89e20] + Distributed 
  [b77e0a4c] + InteractiveUtils 
  [76f85450] + LibGit2 
  [8f399da3] + Libdl 
  [56ddb016] + Logging 
  [d6f4376e] + Markdown 
  [44cfe95a] + Pkg 
  [de0858da] + Printf 
  [3fa0cd96] + REPL 
  [9a3f8284] + Random 
  [ea8e919c] + SHA 
  [9e88b42a] + Serialization 
  [6462fe0b] + Sockets 
  [8dfed614] + Test 
  [cf7118a7] + UUIDs 
  [4ec0a83e] + Unicode 
   Building MPI → `~/.julia/packages/MPI/ZfFyE/deps/build.log`
```

**Comment by simonbyrne** [comment]
It seems that the issue is that `dlopen` in Julia 1.4 will no longer search `/usr/local/lib`: https://github.com/JuliaLang/julia/issues/35548#issuecomment-624449244

#367 will fix this by bundling MPI.

**Comment by simonbyrne** [comment]
You should also be able to set `ENV["JULIA_MPI_PATH"]="/usr/local"`.

**Comment by simonbyrne** [comment]
Fixed by #367.
